cve_bin_tool/helper_script.py

```diff
@@ -251,43 +279,129 @@ def output(self):
         """
 
         # output: filenames, that we search for binary strings
-        print("FILENAME_PATTERNS = [")
+        print("\tFILENAME_PATTERNS = [")
         for filename in self.filename_pattern:
             if self.product == filename:
                 rprint(
-                    f'\t[cyan]r"{filename}"[/], <--- this is a really common filename pattern'
+                    f'\t\t[cyan]r"{filename}"[/], <--- this is a really common filename pattern'
                 )
             elif self.product in filename:
                 if ".so" in filename:
-                    rprint(f'\t[green]r"{filename}"[/],')
+                    rprint(f'\t\t[green]r"{filename}"[/],')
                 else:
                     rprint(
-                        f'\t[bright_green]r"{filename}"[/], <--- you could just use "{self.product}" to match this file'
+                        f'\t\t[bright_green]r"{filename}"[/], <--- you could just use "{self.product}" to match this file'
                     )  # to single-handedly match filenames of type varnishd, varnishlog, varnishtop, etc.
             else:
-                rprint(f'\t[green]r"{filename}"[/],')
-        print("]")
+                rprint(f'\t\t[green]r"{filename}"[/],')
+        print("\t]")
 
         # output: version-strings
-        print("VERSION_PATTERNS = [")
+        print("\tVERSION_PATTERNS = [")
         for version_string in self.version_pattern:
-            rprint(f'\t[green]r"{version_string}"[/],')
-        print("]")
+            rprint(f'\t\t[green]r"{version_string}"[/],')
+        print("\t]")
 
         # output: vendor-product pair
-        print("VENDOR_PRODUCT = ", end="")
+        print("\tVENDOR_PRODUCT = ", end="")
         rprint(self.vendor_product)
 
         self.CONSOLE.rule()
 
 
-def main(filenames, product_name=None):
+def main(argv=None):
+
+    argv = argv or sys.argv
+
+    # Reset logger level to warning
+    LOGGER.setLevel("WARNING")
+
+    parser = argparse.ArgumentParser(
+        prog="helper-script",
+        description=textwrap.dedent(
+            """
+                Helps contributors who want to write a new cve-bin-tool checker find common filenames,
+                version strings, and other necessary data for building a binary checker
+                """
+        ),
+    )
+    # scan directory args
+    parser.add_argument(
+        "filenames",
+        help="files to scan",
+        nargs="+",
+    )
+
+    # product-name args
+    parser.add_argument(
+        "--product",
+        help="provide product-name that would be searched",
+        dest="product_name",
+        action="store",
+    )
+
+    # version-name args
+    parser.add_argument(
+        "--version",
+        help="provide version that would be searched",
+        dest="version_number",
+        action="store",
+    )
+
+    # # search binary args
+    # parser.add_argument()
```

Comment:
This is there for custom string search in the binary.  (I would try to add it, but if I'm not able to do it, then I'll clean this part)